src/huggingface_hub/inference_api.py

```diff
@@ -136,28 +145,56 @@ def __init__(
         self.api_url = f"{ENDPOINT}/pipeline/{self.task}/{repo_id}"
 
     def __repr__(self):
-        items = (f"{k}='{v}'" for k, v in self.__dict__.items())
-        return f"{self.__class__.__name__}({', '.join(items)})"
+        # Do not add headers to repr to avoid leaking token.
+        return (
+            f"InferenceAPI(api_url='{self.api_url}', task='{self.task}',"
+            f" options={self.options})"
+        )
 
     def __call__(
         self,
         inputs: Optional[Union[str, Dict, List[str], List[List[str]]]] = None,
         params: Optional[Dict] = None,
         data: Optional[bytes] = None,
-    ):
+        raw_response: bool = False,
```

Comment:
Maybe this is a good opportunity to add some docstring for this and its arguments

Reply:
I felt when doing so that I was lazy here :innocent:

Reply:
Done in https://github.com/huggingface/huggingface_hub/pull/1270/commits/7ceef22e15fa4ff439168ad87535d302476d382b.